networking/src/NTP/Client.hs

```diff
@@ -146,14 +161,22 @@ startSend :: NtpMonad m => [SockAddr] -> NtpClient m -> m ()
 startSend addrs cli = do
     let timeout = ntpResponseTimeout (ncSettings cli)
     let poll    = ntpPollDelay (ncSettings cli)
-    logDebug "Sending requests"
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Just []
-    () <$ threadDelay timeout `race` forConcurrently addrs (flip doSend cli)
 
-    logDebug "Collecting responses"
-    handleCollectedResponses cli
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Nothing
-    liftIO $ threadDelay (poll - timeout)
+    _ <- concurrently (threadDelay poll) $ do
```

Comment:
This is a last to finish syncronisation, so it will always wait for the full time. Doesn't seem likely that this is what we want.

Reply:
So why are we waiting here at all?

This whole pattern is confused. Are we polling on a regular interval and so the API just picks up the current best known info, or are we doing a single poll when the API is called? Currently it does a mish-mash of both. It starts a thread that polls repeatedly, and so tries to wait between polls, but on the other hand the way we use it we only wait for the first set of responses before killing the client.

So I think this poll timeout never really gets used, since the outer (replacement for) `ntpSingleShot` will kill this thread and return as soon as `handleCollectedResponses` is run.

Reply:
That's true. Originally this NTP client was designed in a way where it can work in background and periodically update remote time kept in `TVar`.
Then `ntpSingleShot` was written to request time once on startup.
Then post-startup use of NTP was refused, and NTP client began to be used only in `ntpSingleShot`.
Just recently, `/api/time/difference` endpoint was written, perhaps taking `ntpSingleShot` as an example. instead of bringing polling in background back.

AFAIU, for mentioned endpoint's sake, we want to use this polling after all.